org.eclipse.lsp4e/src/org/eclipse/lsp4e/VersionedEdits.java

```diff
@@ -44,14 +41,19 @@ public VersionedEdits(long version, List<? extends TextEdit> data, IDocument doc
 	 * received the request
 	 */
 	public void apply() throws BadLocationException, ConcurrentModificationException {
-		if (getVersion() != DocumentUtil.getDocumentModificationStamp(this.document)) {
+		if (this.sourceDocumentVersion != DocumentUtil.getDocumentModificationStamp(this.document)) {
 			throw new ConcurrentModificationException();
 		} else {
-			LSPEclipseUtils.applyEdits(this.document, get());
+			LSPEclipseUtils.applyEdits(this.document, data);
 		}
 	}
 
 	public static @NonNull VersionedEdits toVersionedEdits(@NonNull LanguageServerDocumentExecutor executor, List<? extends TextEdit> data) {
```

Comment:
we could now remove this factory method based on the executor, right?

Reply:
Yes, done.